### DISCUSSION TITLE
solution

### DISCUSSION THREAD

**Discussion by usocoder** [sunng87/pgwire#210] [Show and tell]
https://github.com/usocoder/felix/tree/main/twenty-postgres-proxy

Hey there! My name its Ritchie from New Zealand i have developed a program that acts like a security guard for a database I created a new folder called twenty postgres proxy to organize key files. The core of our solution is a Rust-based program that performs several crucial tasks. It listens for incoming database connection attempts, verifies if the user trying to connect has permission, and if authorized, restricts the user to only the sections of the database they’re allowed to access. It also forwards their queries to the actual database and returns the responses. I created a "config.json" file that allows easy customization of where the program should listen for connections, the location of the real database, and who is allowed to connect and what parts of the database they can access.
here is my email for instant contact
fixatechnz@gmail.com

0x819ffeE0e30BBB4309282297C7F26D11a9cD3350
here is my Eth address if this is successful,
or email what payment methods you can do.

**Comment by sunng87** [comment]
hello @usocoder , it's great to hear you build on pgwire. I'm moving this to discussion and I hope can be a foundation for future work by you or by others. 

Actually a postgres proxy should have way more features than the current state of your project. But I still encourage you keep working on it. Let me know if you need more ideas or help.
